From 391a82e13b5400cade99ebd22294b0c2d6ca75ed Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Mon, 22 Dec 2014 02:43:11 -0500
Subject: [PATCH] Build only the unzip applet

Also, add a new Android.ndk.mk for building with the NDK.
---
 Android.ndk.mk            |  80 +++++++
 busybox-unziponly.config  | 549 ++++++++++++++++++++++++++++++++++++++++++++++
 busybox-unziponly.links   |   1 +
 busybox-unziponly.sources |  26 +++
 4 files changed, 656 insertions(+)
 create mode 100644 Android.ndk.mk
 create mode 100644 busybox-unziponly.config
 create mode 100644 busybox-unziponly.links
 create mode 100644 busybox-unziponly.sources

diff --git a/Android.ndk.mk b/Android.ndk.mk
new file mode 100644
index 0000000..bad6f2a
--- /dev/null
+++ b/Android.ndk.mk
@@ -0,0 +1,80 @@
+LOCAL_PATH := $(call my-dir)
+BB_PATH := $(LOCAL_PATH)
+
+BUSYBOX_CONFIG := unziponly
+
+# Bionic Branches Switches (GB/ICS/L)
+BIONIC_ICS := false
+BIONIC_L := true
+
+#####################################################################
+
+BUSYBOX_CONFIG_DIR := $(BUSYBOX_CONFIG)_$(TARGET_ARCH_ABI)
+
+busybox_prepare_config := $(BUSYBOX_CONFIG_DIR)/.config
+$(busybox_prepare_config): busybox-$(BUSYBOX_CONFIG).config
+	@echo "Prepare config for busybox binary"
+	@rm -rf $(BUSYBOX_CONFIG_DIR)
+	@rm -f $(shell find $(abspath $(call intermediates-dir-for,EXECUTABLES,busybox)) -name "*.o")
+	@mkdir -p $(@D)
+	@cat $^ > $@ && echo "CONFIG_CROSS_COMPILER_PREFIX=\"$(TOOLCHAIN_PREFIX)\"" >> $@
+	+make -C $(BB_PATH) prepare O=$(@D)
+
+busybox_prepare: $(busybox_prepare_config)
+
+#####################################################################
+
+BUSYBOX_SRC_FILES = \
+	$(shell cat busybox-$(BUSYBOX_CONFIG).sources)
+
+BUSYBOX_C_INCLUDES = \
+	$(BUSYBOX_CONFIG_DIR)/include \
+	include \
+	libbb
+
+BUSYBOX_CFLAGS = \
+	-Werror=implicit \
+	-Wno-clobbered \
+	-DNDEBUG \
+	-DANDROID \
+	-fno-strict-aliasing \
+	-fno-builtin-stpcpy \
+	-include $(BUSYBOX_CONFIG_DIR)/include/autoconf.h \
+	-DBB_VER=\"UNZIPONLY\" \
+	-DBB_BT=AUTOCONF_TIMESTAMP
+
+ifeq ($(BIONIC_L),true)
+    BUSYBOX_CFLAGS += -DBIONIC_L
+    BUSYBOX_AFLAGS += -DBIONIC_L
+    # include changes for ICS/JB/KK
+    BIONIC_ICS := true
+endif
+
+ifeq ($(BIONIC_ICS),true)
+    BUSYBOX_CFLAGS += -DBIONIC_ICS
+endif
+
+# Shared Busybox
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := $(BUSYBOX_SRC_FILES)
+LOCAL_C_INCLUDES := $(BUSYBOX_C_INCLUDES)
+LOCAL_CFLAGS := $(BUSYBOX_CFLAGS)
+LOCAL_ASFLAGS := $(BUSYBOX_AFLAGS)
+LOCAL_MODULE := busybox_shared
+LOCAL_SHARED_LIBRARIES := libc libm
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+include $(BUILD_EXECUTABLE)
+
+# Static Busybox
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := $(BUSYBOX_SRC_FILES)
+LOCAL_C_INCLUDES := $(BUSYBOX_C_INCLUDES)
+LOCAL_CFLAGS := $(BUSYBOX_CFLAGS)
+LOCAL_ASFLAGS := $(BUSYBOX_AFLAGS)
+LOCAL_LDFLAGS := -static
+LOCAL_MODULE := busybox_static
+LOCAL_STATIC_LIBRARIES := libc libm
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+include $(BUILD_EXECUTABLE)
diff --git a/busybox-unziponly.config b/busybox-unziponly.config
new file mode 100644
index 0000000..46d2089
--- /dev/null
+++ b/busybox-unziponly.config
@@ -0,0 +1,549 @@
+CONFIG_HAVE_DOT_CONFIG=y
+
+#
+# Busybox Settings
+#
+
+#
+# General Configuration
+#
+CONFIG_DESKTOP=y
+CONFIG_EXTRA_COMPAT=n
+CONFIG_INCLUDE_SUSv2=n
+# CONFIG_USE_PORTABLE_CODE is not set
+CONFIG_PLATFORM_LINUX=y
+CONFIG_FEATURE_BUFFERS_USE_MALLOC=y
+# CONFIG_FEATURE_BUFFERS_GO_ON_STACK is not set
+# CONFIG_FEATURE_BUFFERS_GO_IN_BSS is not set
+CONFIG_SHOW_USAGE=y
+CONFIG_FEATURE_VERBOSE_USAGE=y
+CONFIG_FEATURE_COMPRESS_USAGE=y
+CONFIG_FEATURE_INSTALLER=n
+CONFIG_INSTALL_NO_USR=y
+CONFIG_LOCALE_SUPPORT=n
+CONFIG_UNICODE_SUPPORT=y
+CONFIG_FEATURE_CHECK_UNICODE_IN_ENV=n
+CONFIG_SUBST_WCHAR=63
+CONFIG_LAST_SUPPORTED_WCHAR=0
+CONFIG_UNICODE_COMBINING_WCHARS=n
+CONFIG_UNICODE_WIDE_WCHARS=y
+CONFIG_UNICODE_BIDI_SUPPORT=n
+CONFIG_UNICODE_PRESERVE_BROKEN=y
+CONFIG_LONG_OPTS=y
+CONFIG_FEATURE_DEVPTS=y
+CONFIG_FEATURE_CLEAN_UP=n
+CONFIG_FEATURE_UTMP=n
+CONFIG_FEATURE_WTMP=n
+CONFIG_FEATURE_PIDFILE=n
+CONFIG_FEATURE_SUID=n
+CONFIG_SELINUX=n
+CONFIG_FEATURE_PREFER_APPLETS=n
+CONFIG_BUSYBOX_EXEC_PATH="/proc/self/exe"
+CONFIG_FEATURE_SYSLOG=n
+CONFIG_FEATURE_HAVE_RPC=n
+
+#
+# Build Options
+#
+# CONFIG_STATIC is not set
+# CONFIG_PIE is not set
+# CONFIG_NOMMU is not set
+# CONFIG_BUILD_LIBBUSYBOX is not set
+# CONFIG_FEATURE_INDIVIDUAL is not set
+# CONFIG_FEATURE_SHARED_BUSYBOX is not set
+# CONFIG_LFS is not set
+CONFIG_SYSROOT=""
+CONFIG_EXTRA_CFLAGS="-Os -fno-short-enums -fgcse-after-reload -frerun-cse-after-loop -frename-registers"
+CONFIG_EXTRA_LDFLAGS=""
+CONFIG_EXTRA_LDLIBS=""
+
+#
+# Debugging Options
+#
+# CONFIG_DEBUG is not set
+# CONFIG_DEBUG_PESSIMIZE is not set
+# CONFIG_WERROR is not set
+CONFIG_NO_DEBUG_LIB=y
+# CONFIG_DMALLOC is not set
+# CONFIG_EFENCE is not set
+
+#
+# Installation Options ("make install" behavior)
+#
+CONFIG_INSTALL_APPLET_SYMLINKS=y
+# CONFIG_INSTALL_APPLET_HARDLINKS is not set
+# CONFIG_INSTALL_APPLET_SCRIPT_WRAPPERS is not set
+# CONFIG_INSTALL_APPLET_DONT is not set
+# CONFIG_INSTALL_SH_APPLET_SYMLINK is not set
+# CONFIG_INSTALL_SH_APPLET_HARDLINK is not set
+# CONFIG_INSTALL_SH_APPLET_SCRIPT_WRAPPER is not set
+CONFIG_PREFIX="./_install"
+
+#
+# Busybox Library Tuning
+#
+CONFIG_FEATURE_SYSTEMD=n
+CONFIG_FEATURE_RTMINMAX=y
+CONFIG_PASSWORD_MINLEN=6
+CONFIG_MD5_SMALL=1
+CONFIG_SHA3_SMALL=1
+CONFIG_FEATURE_FAST_TOP=n
+CONFIG_FEATURE_ETC_NETWORKS=n
+CONFIG_FEATURE_USE_TERMIOS=y
+CONFIG_FEATURE_EDITING=n
+CONFIG_FEATURE_NON_POSIX_CP=n
+CONFIG_FEATURE_VERBOSE_CP_MESSAGE=n
+CONFIG_FEATURE_COPYBUF_KB=4
+CONFIG_FEATURE_SKIP_ROOTFS=y
+CONFIG_MONOTONIC_SYSCALL=y
+CONFIG_IOCTL_HEX2STR_ERROR=y
+CONFIG_FEATURE_HWIB=n
+
+#
+# Applets
+#
+
+#
+# Archival Utilities
+#
+CONFIG_FEATURE_SEAMLESS_XZ=n
+CONFIG_FEATURE_SEAMLESS_LZMA=n
+CONFIG_FEATURE_SEAMLESS_BZ2=n
+CONFIG_FEATURE_SEAMLESS_GZ=n
+CONFIG_FEATURE_SEAMLESS_Z=n
+CONFIG_AR=n
+CONFIG_UNCOMPRESS=n
+CONFIG_GUNZIP=n
+CONFIG_BUNZIP2=n
+CONFIG_UNLZMA=n
+CONFIG_LZMA=n
+CONFIG_UNXZ=n
+CONFIG_XZ=n
+CONFIG_BZIP2=n
+CONFIG_CPIO=n
+CONFIG_DPKG=n
+CONFIG_DPKG_DEB=n
+CONFIG_GZIP=n
+CONFIG_LZOP=n
+CONFIG_RPM=n
+CONFIG_RPM2CPIO=n
+CONFIG_TAR=n
+CONFIG_UNZIP=y
+
+#
+# Coreutils
+#
+CONFIG_BASENAME=n
+CONFIG_CAT=n
+CONFIG_DATE=n
+CONFIG_HOSTID=n
+CONFIG_ID=n
+CONFIG_GROUPS=n
+CONFIG_TEST=n
+CONFIG_TOUCH=n
+CONFIG_TR=n
+CONFIG_BASE64=n
+# CONFIG_WHO is not set
+# CONFIG_USERS is not set
+CONFIG_CAL=n
+CONFIG_CATV=n
+CONFIG_CHGRP=n
+CONFIG_CHMOD=n
+CONFIG_CHOWN=n
+CONFIG_CHROOT=n
+CONFIG_CKSUM=n
+CONFIG_COMM=n
+CONFIG_CP=n
+CONFIG_CUT=n
+CONFIG_DD=n
+CONFIG_DF=n
+CONFIG_DIRNAME=n
+CONFIG_DOS2UNIX=n
+CONFIG_UNIX2DOS=n
+CONFIG_DU=n
+CONFIG_ECHO=n
+CONFIG_ENV=n
+CONFIG_EXPAND=n
+CONFIG_EXPR=n
+CONFIG_FALSE=n
+CONFIG_FOLD=n
+CONFIG_FSYNC=n
+CONFIG_HEAD=n
+CONFIG_INSTALL=n
+CONFIG_LN=n
+CONFIG_LOGNAME=n
+CONFIG_LS=n
+CONFIG_MD5SUM=n
+CONFIG_MKDIR=n
+CONFIG_MKFIFO=n
+CONFIG_MKNOD=n
+CONFIG_MV=n
+CONFIG_NICE=n
+CONFIG_NOHUP=n
+CONFIG_OD=n
+CONFIG_PRINTENV=n
+CONFIG_PRINTF=n
+CONFIG_PWD=n
+CONFIG_READLINK=n
+CONFIG_REALPATH=n
+CONFIG_RM=n
+CONFIG_RMDIR=n
+CONFIG_SEQ=n
+CONFIG_SHA1SUM=n
+CONFIG_SHA256SUM=n
+CONFIG_SHA512SUM=n
+CONFIG_SHA3SUM=n
+CONFIG_SLEEP=n
+CONFIG_SORT=n
+CONFIG_SPLIT=n
+CONFIG_STAT=n
+CONFIG_STTY=n
+CONFIG_SUM=n
+CONFIG_SYNC=n
+CONFIG_TAC=n
+CONFIG_TAIL=n
+CONFIG_TEE=n
+CONFIG_TRUE=n
+CONFIG_TTY=n
+CONFIG_UNAME=n
+CONFIG_UNEXPAND=n
+CONFIG_UNIQ=n
+CONFIG_USLEEP=n
+CONFIG_UUDECODE=n
+CONFIG_UUENCODE=n
+CONFIG_WC=n
+CONFIG_WHOAMI=n
+CONFIG_YES=n
+
+#
+# Console Utilities
+#
+CONFIG_CHVT=n
+CONFIG_FGCONSOLE=n
+CONFIG_CLEAR=n
+CONFIG_DEALLOCVT=n
+CONFIG_DUMPKMAP=n
+CONFIG_KBD_MODE=n
+CONFIG_LOADFONT=n
+CONFIG_LOADKMAP=n
+CONFIG_OPENVT=n
+CONFIG_RESET=n
+CONFIG_RESIZE=n
+CONFIG_SETCONSOLE=n
+CONFIG_SETFONT=n
+CONFIG_SETKEYCODES=n
+CONFIG_SETLOGCONS=n
+CONFIG_SHOWKEY=n
+
+#
+# Debian Utilities
+#
+CONFIG_MKTEMP=n
+CONFIG_PIPE_PROGRESS=n
+CONFIG_RUN_PARTS=n
+CONFIG_START_STOP_DAEMON=n
+CONFIG_WHICH=n
+
+#
+# Editors
+#
+CONFIG_AWK=n
+CONFIG_CMP=n
+CONFIG_DIFF=n
+CONFIG_ED=n
+CONFIG_PATCH=n
+CONFIG_SED=n
+CONFIG_VI=n
+
+#
+# Finding Utilities
+#
+CONFIG_FIND=n
+CONFIG_GREP=n
+CONFIG_XARGS=n
+
+#
+# Init Utilities
+#
+CONFIG_BOOTCHARTD=n
+CONFIG_HALT=n
+CONFIG_INIT=n
+CONFIG_MESG=n
+
+#
+# Login/Password Management Utilities
+#
+CONFIG_ADD_SHELL=n
+CONFIG_REMOVE_SHELL=n
+# CONFIG_FEATURE_SHADOWPASSWDS is not set
+# CONFIG_USE_BB_PWD_GRP is not set
+# CONFIG_USE_BB_SHADOW is not set
+CONFIG_USE_BB_CRYPT=y
+CONFIG_USE_BB_CRYPT_SHA=y
+CONFIG_ADDUSER=n
+CONFIG_ADDGROUP=n
+CONFIG_DELUSER=n
+CONFIG_DELGROUP=n
+CONFIG_GETTY=n
+CONFIG_LOGIN=n
+CONFIG_PASSWD=n
+CONFIG_CRYPTPW=n
+CONFIG_CHPASSWD=n
+CONFIG_SU=n
+CONFIG_SULOGIN=n
+CONFIG_VLOCK=n
+
+#
+# Linux Ext2 FS Progs
+#
+CONFIG_CHATTR=n
+CONFIG_FSCK=n
+CONFIG_LSATTR=n
+CONFIG_TUNE2FS=n
+
+#
+# Linux Module Utilities
+#
+CONFIG_MODINFO=n
+CONFIG_MODPROBE_SMALL=n
+CONFIG_INSMOD=n
+CONFIG_RMMOD=n
+CONFIG_LSMOD=n
+CONFIG_MODPROBE=n
+CONFIG_DEPMOD=n
+
+#
+# Linux System Utilities
+#
+CONFIG_BLOCKDEV=n
+CONFIG_FSTRIM=n
+CONFIG_MDEV=n
+CONFIG_REV=n
+CONFIG_ACPID=n
+CONFIG_BLKID=n
+CONFIG_DMESG=n
+CONFIG_FBSET=n
+CONFIG_FDFLUSH=n
+CONFIG_FDFORMAT=n
+CONFIG_FDISK=n
+CONFIG_FINDFS=n
+CONFIG_FLOCK=n
+CONFIG_FREERAMDISK=n
+CONFIG_FSCK_MINIX=n
+CONFIG_MKFS_EXT2=n
+CONFIG_MKFS_MINIX=n
+CONFIG_MKFS_REISER=n
+CONFIG_MKFS_VFAT=n
+CONFIG_GETOPT=n
+CONFIG_HEXDUMP=n
+# CONFIG_HD is not set
+CONFIG_HWCLOCK=n
+CONFIG_IPCRM=n
+CONFIG_IPCS=n
+CONFIG_LOSETUP=n
+CONFIG_LSPCI=n
+CONFIG_LSUSB=n
+CONFIG_MKSWAP=n
+CONFIG_MORE=n
+CONFIG_MOUNT=n
+CONFIG_PIVOT_ROOT=n
+CONFIG_RDATE=n
+CONFIG_RDEV=n
+CONFIG_READPROFILE=n
+CONFIG_RTCWAKE=n
+CONFIG_SCRIPT=n
+CONFIG_SCRIPTREPLAY=n
+CONFIG_SETARCH=n
+CONFIG_SWAPONOFF=n
+CONFIG_SWITCH_ROOT=n
+CONFIG_UMOUNT=n
+
+#
+# Miscellaneous Utilities
+#
+CONFIG_CONSPY=n
+CONFIG_LESS=n
+CONFIG_NANDWRITE=n
+CONFIG_NANDDUMP=n
+CONFIG_RFKILL=n
+CONFIG_SETSERIAL=n
+CONFIG_UBIATTACH=n
+CONFIG_UBIDETACH=n
+CONFIG_UBIMKVOL=n
+CONFIG_UBIRMVOL=n
+CONFIG_UBIRSVOL=n
+CONFIG_UBIUPDATEVOL=n
+# CONFIG_WALL is not set
+CONFIG_ADJTIMEX=n
+CONFIG_BBCONFIG=n
+CONFIG_BEEP=n
+CONFIG_CHAT=n
+CONFIG_CHRT=n
+CONFIG_CROND=n
+CONFIG_CRONTAB=n
+CONFIG_DC=n
+CONFIG_DEVFSD=n
+CONFIG_FEATURE_DEVFS=n
+CONFIG_DEVMEM=n
+CONFIG_EJECT=n
+CONFIG_FBSPLASH=n
+CONFIG_FLASHCP=n
+CONFIG_FLASH_LOCK=n
+CONFIG_FLASH_UNLOCK=n
+CONFIG_FLASH_ERASEALL=n
+CONFIG_IONICE=n
+CONFIG_INOTIFYD=n
+# CONFIG_LAST is not set
+CONFIG_HDPARM=n
+CONFIG_MAKEDEVS=n
+CONFIG_MAN=n
+CONFIG_MICROCOM=n
+CONFIG_MOUNTPOINT=n
+CONFIG_MT=n
+CONFIG_RAIDAUTORUN=n
+CONFIG_READAHEAD=n
+# CONFIG_RUNLEVEL is not set
+CONFIG_RX=n
+CONFIG_SETSID=n
+CONFIG_STRINGS=n
+CONFIG_TASKSET=n
+CONFIG_TIME=n
+CONFIG_TIMEOUT=n
+CONFIG_TTYSIZE=n
+CONFIG_VOLNAME=n
+CONFIG_WATCHDOG=n
+
+#
+# Networking Utilities
+#
+CONFIG_NAMEIF=n
+CONFIG_NBDCLIENT=n
+CONFIG_NC=n
+CONFIG_PING=n
+CONFIG_WHOIS=n
+CONFIG_FEATURE_IPV6=n
+CONFIG_FEATURE_UNIX_LOCAL=n
+CONFIG_VERBOSE_RESOLUTION_ERRORS=n
+CONFIG_ARP=n
+CONFIG_ARPING=n
+CONFIG_BRCTL=n
+CONFIG_DNSD=n
+CONFIG_ETHER_WAKE=n
+CONFIG_FAKEIDENTD=n
+CONFIG_FTPD=n
+CONFIG_FTPGET=n
+CONFIG_FTPPUT=n
+CONFIG_HOSTNAME=n
+CONFIG_HTTPD=n
+CONFIG_IFCONFIG=n
+CONFIG_IFENSLAVE=n
+CONFIG_IFPLUGD=n
+CONFIG_IFUPDOWN=n
+CONFIG_INETD=n
+CONFIG_IP=n
+CONFIG_IPCALC=n
+CONFIG_NETSTAT=n
+CONFIG_NSLOOKUP=n
+CONFIG_NTPD=n
+CONFIG_PSCAN=n
+CONFIG_ROUTE=n
+CONFIG_SLATTACH=n
+CONFIG_TCPSVD=n
+CONFIG_TELNET=n
+CONFIG_TELNETD=n
+CONFIG_TFTP=n
+CONFIG_TFTPD=n
+CONFIG_TRACEROUTE=n
+CONFIG_TUNCTL=n
+CONFIG_UDHCPC6=n
+CONFIG_UDHCPD=n
+CONFIG_UDHCPC=n
+CONFIG_UDPSVD=n
+CONFIG_VCONFIG=n
+CONFIG_WGET=n
+CONFIG_ZCIP=n
+
+#
+# Print Utilities
+#
+CONFIG_LPD=n
+CONFIG_LPR=n
+CONFIG_LPQ=n
+
+#
+# Mail Utilities
+#
+CONFIG_MAKEMIME=n
+CONFIG_POPMAILDIR=n
+CONFIG_REFORMIME=n
+CONFIG_SENDMAIL=n
+
+#
+# Process Utilities
+#
+CONFIG_IOSTAT=n
+CONFIG_LSOF=n
+CONFIG_MPSTAT=n
+CONFIG_NMETER=n
+CONFIG_PMAP=n
+CONFIG_POWERTOP=n
+CONFIG_PSTREE=n
+CONFIG_PWDX=n
+CONFIG_SMEMCAP=n
+CONFIG_TOP=n
+CONFIG_UPTIME=n
+CONFIG_FREE=n
+CONFIG_FUSER=n
+CONFIG_KILL=n
+CONFIG_PGREP=n
+CONFIG_PIDOF=n
+CONFIG_PKILL=n
+CONFIG_PS=n
+CONFIG_RENICE=n
+CONFIG_BB_SYSCTL=n
+CONFIG_WATCH=n
+
+#
+# Runit Utilities
+#
+CONFIG_RUNSV=n
+CONFIG_RUNSVDIR=n
+CONFIG_SV=n
+CONFIG_SVLOGD=n
+CONFIG_CHPST=n
+CONFIG_SETUIDGID=n
+CONFIG_ENVUIDGID=n
+CONFIG_ENVDIR=n
+CONFIG_SOFTLIMIT=n
+
+#
+# SELinux Utilities
+#
+CONFIG_CHCON=n
+CONFIG_GETENFORCE=n
+CONFIG_GETSEBOOL=n
+CONFIG_MATCHPATHCON=n
+CONFIG_RESTORECON=n
+CONFIG_RUNCON=n
+CONFIG_SELINUXENABLED=n
+CONFIG_SETENFORCE=n
+CONFIG_SETFILES=n
+CONFIG_SETSEBOOL=n
+CONFIG_SESTATUS=n
+
+#
+# Shells
+#
+CONFIG_ASH=n
+CONFIG_CTTYHACK=n
+CONFIG_HUSH=n
+CONFIG_MSH=n
+CONFIG_FEATURE_SH_IS_NONE=y
+CONFIG_FEATURE_BASH_IS_NONE=y
+
+#
+# System Logging Utilities
+#
+CONFIG_SYSLOGD=n
+CONFIG_KLOGD=n
+CONFIG_LOGGER=n
diff --git a/busybox-unziponly.links b/busybox-unziponly.links
new file mode 100644
index 0000000..8452996
--- /dev/null
+++ b/busybox-unziponly.links
@@ -0,0 +1 @@
+/bin/unzip
diff --git a/busybox-unziponly.sources b/busybox-unziponly.sources
new file mode 100644
index 0000000..dcbe1fb
--- /dev/null
+++ b/busybox-unziponly.sources
@@ -0,0 +1,26 @@
+android/libc/__set_errno.c
+
+archival/unzip.c
+archival/libarchive/decompress_bunzip2.c
+archival/libarchive/decompress_gunzip.c
+archival/libarchive/find_list_entry.c
+archival/libarchive/open_transformer.c
+
+libbb/appletlib.c
+libbb/chomp.c
+libbb/copyfd.c
+libbb/crc32.c
+libbb/default_error_retval.c
+libbb/full_write.c
+libbb/last_char_is.c
+libbb/llist.c
+libbb/make_directory.c
+libbb/messages.c
+libbb/perror_msg.c
+libbb/read.c
+libbb/read_printf.c
+libbb/safe_write.c
+libbb/verror_msg.c
+libbb/xfunc_die.c
+libbb/xfuncs.c
+libbb/xfuncs_printf.c
-- 
2.2.1

